src/devices/cpu/m6502/gew12.cpp

```diff
@@ -0,0 +1,229 @@
+// license:BSD-3-Clause
+// copyright-holders:Devin Acker
+/***************************************************************************
+
+    gew7.cpp
+
+    Yamaha GEW2 (65c02-based)
+
+***************************************************************************/
+
+#include "emu.h"
+#include "gew12.h"
+
+#include "m6502mcu.ipp"
+
+DEFINE_DEVICE_TYPE(GEW12, gew12_device, "gew12", "Yamaha YMW728-F (GEW12)")
+
+gew12_device::gew12_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock)
+	: m6502_mcu_device_base<m65c02_device>(mconfig, GEW12, tag, owner, clock)
+	, device_mixer_interface(mconfig, *this, 2)
+	, m_in_cb(*this), m_out_cb(*this)
+	, m_rom(*this, DEVICE_SELF)
+	, m_bank(*this, "bank%u", 0U)
+{
+	program_config.m_internal_map = address_map_constructor(FUNC(gew12_device::internal_map), this);
+}
+
+
+void gew12_device::device_add_mconfig(machine_config &config)
+{
+	// TODO: PCM, UART
+}
+
+void gew12_device::device_start()
+{
+	m65c02_device::device_start();
+
+	m_in_cb.resolve_all_safe(0);
+	m_out_cb.resolve_all_safe();
+
+	m_bank[0]->configure_entries(0, m_rom->bytes() >> 14, m_rom->base(), 1 << 14);
+	m_bank[1]->configure_entries(0, m_rom->bytes() >> 14, m_rom->base(), 1 << 14);
+	m_bank_mask = (m_rom->bytes() >> 14) - 1;
+
+	m_timer_base[0] = m_timer_base[1] = 0;
+
+	memset(m_port_data, 0, sizeof m_port_data);
+	memset(m_port_ddr, 0, sizeof m_port_ddr);
```

Comment:
Please use `std::fill` or `std::fill_n for these (e.g. `std::fill(std::begin(m_port_data), std::end(m_port_data), 0);` or `std::fill_n(std::begin(m_port_data), std::size(m_port_data), 0);`) – it’s less error-prone than `memset`.